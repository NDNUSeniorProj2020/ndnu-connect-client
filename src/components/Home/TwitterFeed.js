import React, { useEffect } from 'react';
import { Card } from 'antd';

export default function TwitterFeed() {
	useEffect(() => {
		const script = document.createElement("script");
		script.src = 'https://platform.twitter.com/widgets.js';
		document.getElementsByClassName("twitter-embed")[0].appendChild(script);
	}, []);

	return (
		<Card className="tutors-card-container">
			<section className={'twitter-container'}>
				<div className={'twitter-embed'}>
					<a className="twitter-timeline" data-width="400" data-height="400" data-dnt="true"
						href="https://twitter.com/NotreDameCA?ref_src=twsrc%5Etfw">Tweets by NotreDameCA</a>
				</div>
			</section>
		</Card>
	);
}
