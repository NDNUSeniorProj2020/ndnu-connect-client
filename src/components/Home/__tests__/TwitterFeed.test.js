import React from 'react';
import renderer from 'react-test-renderer';

import TwitterFeed from '../TwitterFeed';

describe('Test TwitterFeed component', () => {
	describe('Snapshot tests', () => {
		it('should render without crashing.', () => {
			const tree = renderer.create(<TwitterFeed />);
			expect(tree).toMatchSnapshot();
		});
	});
});