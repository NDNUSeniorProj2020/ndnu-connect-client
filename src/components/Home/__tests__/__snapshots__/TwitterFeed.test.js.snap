// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for TwitterFeed component snapshot tests should render without crashing. 1`] = `
<div
  className="ant-card tutors-card-container ant-card-bordered"
>
  <div
    className="ant-card-body"
    style={Object {}}
  >
    <section
      className="twitter-container"
    >
      <div
        className="twitter-embed"
      >
        <a
          className="twitter-timeline"
          data-dnt="true"
          data-height="400"
          data-width="400"
          href="https://twitter.com/NotreDameCA?ref_src=twsrc%5Etfw"
        >
          Tweets by NotreDameCA
        </a>
      </div>
    </section>
  </div>
</div>
`;
